### DISCUSSION TITLE
Recommended way to theme AppShell border in v7

### DISCUSSION THREAD

**Discussion by KevinSchiener** [mantinedev/mantine#5017] [Q&A] [answered]
Hello,

we're currently migrating to Mantine 7.x with a custom theme and I was wondering what the intended way to style, for example, the AppShell border is. We tried the following methods so far:

1. Override each border-color CSS property of the subcomponents (aside, header, nav, footer)
```TypeScript
AppShell.extend({
            classNames: {
                root: classes.appShellRoot,
                main: classes.appShellMain,
                header: classes.appShellPanel,
                footer: classes.appShellFooter,
                aside: classes.appShellPanel,
                navbar: classes.appShellPanel,
            },
        })
```
2. Override the `--_app-shell-border-color` variable in the AppShell's `root` CSS class.
3. Override the `--_app-shell-border-color` in the `vars` of the `AppShell.extend(...)`

While 1. seems to lose the indirection via the shared variable, the `--_app-shell-border-color` variable is not mentioned as any sort of official API.

Which way would you recommend? Can we use the `_component-...` variables in overrides or are those implementation details?

Thank you!

**Comment by rtivital** [comment, marked as answer]
I recommend overriding styles with public properties of Styles API. `--_app-shell-border-color` and other CSS variables that start with `--_` are considered to be a private API – they are not documented and may have breaking changes in minor/patches.

**Reply by KevinSchiener** [reply]
Thanks. Thought as much. We opted to override the classes for now, since there are no variables for this sort of styling yet.
